spark-extension/src/main/scala/org/apache/spark/sql/blaze/BlazeConvertStrategy.scala

```diff
@@ -176,7 +177,8 @@ object BlazeConvertStrategy extends Logging {
         e.setTagValue(convertStrategyTag, AlwaysConvert)
       case e: LocalTableScanExec =>
         e.setTagValue(convertStrategyTag, AlwaysConvert)
-      case e: DataWritingCommandExec if isNative(e.child) =>
+      case e: DataWritingCommandExec
+          if isNative(e.child) || e.child.isInstanceOf[AdaptiveSparkPlanExec] =>
```

Comment:
do we need to check `isNative(e.child.child)` in this situation?

Reply:
> do we need to check `isNative(e.child.child)` in this situation?
+1